hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2413,6 +2415,8 @@ func reconcileAutoScalerDeployment(deployment *appsv1.Deployment, hc *hyperv1.Ho
 		},
 	}
 
+	util.AvailabilityProber(kas.InClusterKASReadyURL(deployment.Namespace, kas.KubeAPIServerPort()), &deployment.Spec.Template.Spec)
```

Comment:
do we need any other than InClusterKASReadyURL to be exported? thinking from a maintenance pov, one this is released changing those would be breaking changes.

Reply:
You mean the port? I did that because its plugged through as a config option in control-plane-operator/controllers/hostedcontrolplane/kas/kubeconfig.go. It ends up being hardcoded, but it looked to me like it was built to potentially be configurable.

Reply:
I think Alberto is referring to https://github.com/openshift/hypershift/pull/517/files#diff-a8bf07a0034d15a8a140a106f8b4c6d0d78a24a52531577a5cd7501682833f1bR31 
You only need the InClusterKASReadyURL function to be public